libraries/libfc/src/crypto/elliptic_r1.cpp

```diff
@@ -521,6 +524,26 @@ namespace fc { namespace crypto { namespace r1 {
         FC_THROW_EXCEPTION( exception, "unable to reconstruct public key from signature" );
     }
 
+    bool public_key::is_canonical( const compact_signature& c ) {
+        int nV = c.data[0];
+        if (nV<27 || nV>=35)
+            FC_THROW_EXCEPTION( exception, "unable to reconstruct public key from signature" );
+
+        ecdsa_sig sig = ECDSA_SIG_new();
+        BIGNUM *r = BN_new(), *s = BN_new();
+        BN_bin2bn(&c.data[1],32,r);
+        BN_bin2bn(&c.data[33],32,s);
+        ECDSA_SIG_set0(sig, r, s);
```

Comment:
afaict all you need from the cluster above is just
```cpp
BIGNUM *s = BN_new();
BN_bin2bn(&c.data[33],32,s);
```
`sig` is not used in the rest of the function, ergo `r` is not used either